src/docbuilder/queue.rs

```diff
@@ -2,52 +2,24 @@
 
 
 use super::DocBuilder;
-use rustc_serialize::json::{Json, Array};
-use hyper;
-use hyper::header::{Accept, qitem};
 use db::connect_db;
 use errors::*;
+use crates_index_diff::{ChangeKind, Index};
 
 
 impl DocBuilder {
     /// Updates crates.io-index repository and adds new crates into build queue
     pub fn get_new_crates(&mut self) -> Result<()> {
         try!(self.load_database_cache());
-
-        let body = {
-            use std::io::Read;
-            let client = hyper::Client::new();
-            let mut res = try!(client.get("https://crates.io/summary")
-                .header(Accept(vec![qitem("application/json".parse().unwrap())]))
-                .send());
-            let mut body = String::new();
-            try!(res.read_to_string(&mut body));
-            body
-        };
-
-        let json = try!(Json::from_str(&body));
-
-        let crates = {
-            let mut crates: Vec<(String, String)> = Vec::new();
-            for section in ["just_updated", "new_crates"].iter() {
-                match json.as_object()
-                    .and_then(|o| o.get(&section[..]))
-                    .and_then(|j| j.as_array())
-                    .map(get_crates_from_array) {
-                    Some(mut c) => crates.append(c.as_mut()),
-                    None => continue,
-                }
-            }
-            crates
-        };
-
         let conn = try!(connect_db());
-        for (name, version) in crates {
-            if self.db_cache.contains(&format!("{}-{}", name, version)[..]) {
+        let index = try!(Index::from_path_or_cloned(&self.options.crates_io_index_path));
+        let changes = try!(index.fetch_changes());
+        for krate in changes.iter().filter(|k| k.kind != ChangeKind::Yanked) {
+            if self.db_cache.contains(&format!("{}-{}", krate.name, krate.version)) {
```

Comment:
I am not quite sure if that is still needed, assuming that the crates.io index will never repeat itself. It might be that the previous implementation needed it to prevent duplicates or unnecessary work.